Bioconductor/sweave2rmd: Conversion of howtogenefinder.Rnw to Rmd

devbird007 opened the pull request:
@Bioconductor/sweave2rmd
I am an Outreachy Applicant, please review my pull request.

@jwokaty Hello, I have converted the howtogenefinder.Rnw file to Rmd. Kindly review my PR.
The PDF and HTML files for comparison:
[PDF](https://github.com/Bioconductor/genefilter/files/10929144/howtogenefinder.pdf)
[HTML Zip](https://github.com/Bioconductor/genefilter/files/10994305/howtogenefinder.zip)

devbird007 committed:
Update DESCRIPTION with BiocStyle, knitr, VignetteBuilder

devbird007 committed:
Convert howtogenefinder to Rmd

devbird007 committed:
Remove howtogenefinder.Rnw

mcarlsn commented on vignettes/howtogenefinder.Rmd:
```diff
@@ -0,0 +1,77 @@
+---
+title: "How to find genes whose expression profile is similar to that of
+specified genes"
+author:  
+- Robert Gentleman
+- Vincent J. Carey
+- Wolfgang Huber
+- Florian Hahne
+date: "January 19, 2023"
+package: genefilter
+vignette: >
+  %\VignetteIndexEntry{How to find genes whose expression profile is similar to that of
+specified genes}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+output:
+  BiocStyle::html_document
+---
```
A few edits to the YAML: 

- Author names should be in "" 
- You can add your name to the author list as the person who converted the vignette (see comment below for where you can add your name) 
- The date line can be changed to "`r format(Sys.time(), '%B %d, %Y')`" 
- Slight edits to the vignette lines, please see below

Suggested changes: 
```
---
title: "How to find genes whose expression profile is similar to that of specified genes"
author: 
- name: "Robert Gentleman"
- name: "Vincent J. Carey"
- name: "Wolfgang Huber" 
- name: "Florian Hahne" 
- name: "Name of person converting the vignette"
  affiliation: "Vignette translation from Sweave to Rmarkdown / HTML"
date: "`r format(Sys.time(), '%B %d, %Y')`"
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{How to find genes whose expression profile is similar to that of specified genes}
  %\VignettePackage{genefilter}
  %\VignetteEncoding{UTF-8}
output:
  BiocStyle::html_document:
    number_sections: yes
    toc: yes
    toc_depth: 4 
--- 
```

devbird007 replied:
I have made the edits exactly as you requested. I also updated the DESCRIPTION file with my name and the role I performed. Can I also add my name as a contributor to future vignettes I may work on?

mcarlsn replied:
Yes, as long as the maintainer agrees then you can use the same formatting I offered (above) for other vignettes that you convert. Checking with the maintainer is especially important when the vignette/package is not maintained by Bioconductor.

devbird007 replied:
Alright, understood.

Also, I have completely corrected all the errors, reviewed my pull request and successfully ran `R CMD build genefilter`

Perhaps when you're free, you could go over my pull request and then merge it, so I would be free to gleefully go ahead and tackle a hard task. :-)

jwokaty replied:
I noticed that the original PDF doesn't list any authors and it's not also metadata in the .Rnw, so we don't really who are the authors. Can I ask you remove Robert, Vince, Florian, and Wolfgang? It's okay to leave yourself as the contributor.

devbird007 replied:
Done so.

mcarlsn commented on vignettes/howtogenefinder.Rmd:
````diff
@@ -0,0 +1,77 @@
+Two other options for `genefinder` are `scale` and `method`. The `scale` option controls the scaling of the rows (this is often desirable), while the `method` option controls the distance measure used between genes. The possible values and their meanings are listed at the end of this document.
+```{r, message = FALSE}
+library("Biobase")
+```
````
Please add a single carriage line between the `library("Biobase")` line and the `library("genefilter)...` chunk

devbird007 replied:
I don't know what a single carriage line is, but I used the `&nbsp;   ` code to add one empty new line in the final html output where you said to add a single carriage line. I also did this for the other comments where you ask for the same thing. Do clarify for me if this is what you meant, thanks.

jwokaty replied:
A carriage return is blank line. If you used a `&nbsp;`, please replace these.

jwokaty replied:
A carriage return is blank line. If you used a `&nbsp;`, please replace these.

devbird007 replied:
Thanks for the correction. I have made the changes now. I also removed the `&emsp;` tags I used in other places in the file.

mcarlsn commented on vignettes/howtogenefinder.Rmd:
````diff
@@ -0,0 +1,77 @@
+names(closeg)
+``` 
+
+The Affymetrix identifiers (since these were originally Affymetrix data) are `31539_r_at`, `31572_at`, `31594_at` and `31658_at`. We can find the nearest genes (by index) for any of these by simply accessing the relevant component of `closeg`.
````
Please add a single carriage line between the passage that begins with "The Affymetrix identifiers..." and the code chunk below it

mcarlsn replied:
Resolved

mcarlsn commented on vignettes/howtogenefinder.Rmd:
````diff
@@ -0,0 +1,77 @@
+The Affymetrix identifiers (since these were originally Affymetrix data) are `31539_r_at`, `31572_at`, `31594_at` and `31658_at`. We can find the nearest genes (by index) for any of these by simply accessing the relevant component of `closeg`.
+```{r}
+closeg$"31539_r_at"
+```
````
Please add a single carriage line between `closeg$"31539_r_at"` and the code chunk below it

mcarlsn commented on vignettes/howtogenefinder.Rmd:
````diff
@@ -0,0 +1,77 @@
+```{r}
+Nms1 <- featureNames(sample.ExpressionSet)[closeg$"31539_r_at"$indices]
+Nms1
+```
````
Add a single carriage line between the code chunk and the passage that begins with "You could then take these names..."

mcarlsn commented on vignettes/howtogenefinder.Rmd:
````diff
@@ -0,0 +1,77 @@
+Nms1 <- featureNames(sample.ExpressionSet)[closeg$"31539_r_at"$indices]
+Nms1
+```
+You could then take these names (from `Nms1`) and the annotate package and explore them further. See the various **HOWTO's** in annotate to see how to further explore your data. Examples include finding and searching all PubMed abstracts associated with these data. Finding and downloading associated sequence information. The data can also be visualized using the *geneplotter* package (again there ...
````
See changes below - I added `r Biocpkg("annotate")` and `r Biocpkg("geneplotter")` for the 2 packages that were called out. Also, I don't think that "HOWTO" is bolded in the original vignette. I removed the formatting for both instances of the word. 

_Updated chunk:_ 

You could then take these names (from `Nms1`) and the `r Biocpkg("annotate")` package and explore them further. See the various HOWTO's in annotate to see how to further explore your data. Examples include finding and searching all PubMed abstracts associated with these data. Finding and downloading associated sequence information. The data can also be visualized using the `r Biocpkg("geneplotter")` package (again there are a number of HOWTO documents there).

devbird007 replied:
Done

mcarlsn commented on vignettes/howtogenefinder.Rmd:
```diff
@@ -0,0 +1,77 @@
+
+**zscore**&emsp;Scaling is done by $(x_{i} − \bar{x})/s_{x}$. Where s~x~ is the standard deviation.  
+
+The method parameter can take the following values:  
```
Please update the formatting for "method" to `method`

devbird007 replied:
Also done. :-)

mcarlsn replied:
Resolved

devbird007 commented:
Hi @jwokaty. After you assign it to me, what do I do next? I have already reviewed it myself with the guide you sent earlier.

mcarlsn commented:
Great work! Please see comments above and checklist below for needed changes. 

- [ ] Remove the `howtogenefilter.Rnw` file - I see that you attempted to remove the .Rmw file but it's still giving us a hard time with R CMD build and showing up in your forked repository. Do you mind checking to see that it has been removed? 
- [ ] `R CMD build` runs with error: "Vignette re-building failed" because it failed "re-building ‘howtogenefilter.Rnw" - it's possible that this will be fixed by removing howtogenefilter.Rnw, which is another needed step to complete converting .Rnw to .Rmd. Please try removing the .Rmw file and rerunning the build statement with this line: `R_LIBS_USER="/mnt/STORE1/lib/R/bioc-release" R CMD build genefilter`
- [ ] Once `R CMD build` runs without error, we can check to make sure that the tarball from `R CMD build` contains the `HTML` using the syntax `tar ztf genefilter.tar.gz | grep 'doc/howtogenefilter'` (@jwokaty flagging to make sure this line is correct for package: genefilter and vignette: howtogenefilter) 
- [ ] As contributor, you can add your name to the author list in the `DESCRIPTION` file using this formatting: `person("First Name", "Last Name", role="ctb", comment="'howtogenefilter' vignette translation from Sweave to Rmarkdown / HTML")`
- [ ] As contributor, you can also add your name to the author list in the vignette's YAML, see above comment for additional guidance

devbird007 committed:
Update DESCRIPTION with BiocStyle, knitr, VignetteBuilder

devbird007 committed:
Convert howtogenefinder.Rnw to Rmd

devbird007 commented:
In the vignette folder where my `howtogenefinder.Rmd` file is located, there are two other Rnw files one of which is `howtogenefilter.Rnw` and it was assigned to my fellow outreachy applicant, she is actively working on it from what I have seen. The other Rnw file is not on the Outreachy Project board, so none of us are working on it. 

To successfully run R CMD build, I had to copy the whole repository to a separate environment, delete the two Rnw files from the vignette folder and try the command.

I added my name in the DESCRIPTION file as `...= " 'howtogenefinder' vignette translation..."` instead. I imagine the one you sent would be better suited for the lady working on the `howtogenefilter.Rnw` vignette. If I am wrong though please correct me haha.

devbird007 committed:
Update DESCRIPTION with BiocStyle, knitr, VignetteBuilder

jwokaty commented:
> To successfully run R CMD build, I had to copy the whole repository to a separate environment, delete the two Rnw files from the vignette folder and try the command.

@devbird007 To run `R CMD build` on the package, you shouldn't have to delete the other 2 Rnw files. Were you getting an error when you tried to run this command? If you were getting an error, can you copy and paste the contents of that error here.

@mcarlsn Syntax is correct. I think the names of the vignettes are too similar so it's confusing. This is for howtogenefinder.Rmd.

devbird007 commented:
> 

@jwokaty This is the error I am getting:
![Screenshot from 2023-03-13 21-05-56](https://user-images.githubusercontent.com/100073682/224820630-71254237-da96-4bbc-b9ed-2402cb34ebc6.png)


If you'll observe, the file I was working on "howtogenefinder.Rmd" builds successfully. But the two other .Rnw files generate errors and stop the process.

However, when I copy this directory to another location where I change the directory name to "test-repository" and I delete the .Rnw files, it builds successfully:
![Screenshot from 2023-03-13 21-09-00](https://user-images.githubusercontent.com/100073682/224820685-bef93ff4-0ae3-4761-813c-1c382c5223ae.png)


I've been stuck at this stage for days now and been unable to work on other vignettes, I believe this is the only obstacle left.

devbird007 committed:
Deleted the &nbsp; tags

devbird007 committed:
Updated howtogenefinder.Rmd file

jwokaty commented:
@devbird007 I think the issue is that you don't have pdflatex on your machine, so it can't create the PDF documents. This is why you're able to build if you delete the .Rnw files. Did you install pandoc? And can I ask what is your operating system? If it is Linux, can you share the results of `sudo apt search texlive`? I think pdflatex gets installed with texlive and I am thinking if I should recommend installing texlive. I generally don't because it's a very large package.

devbird007 commented:
Hi @jwokaty , after running into some errors and trying a lot of things, I eventually discovered that what I need to run is `sudo apt install texlive-fonts-extra`. A truth I gleaned from this site [here](https://latex.org/forum/viewtopic.php?t=25989).

You are correct, it is indeed a large file. I will not be able to install it in a few hours. Once I am done, I will let you know the result.

Thanks for putting me on the right path, so I could work my way through the issue.

devbird007 commented:
Having downloaded the required applications, it runs perfectly now:
![Screenshot from 2023-03-15 03-26-24](https://user-images.githubusercontent.com/100073682/225188957-9ecb61ad-add1-4cf5-a3a7-b6db1daa8f60.png)

I believe I am done now.

devbird007 committed:
Wrapped lines and finalized formatting

jwokaty commented on vignettes/howtogenefinder.Rmd:
````diff
@@ -0,0 +1,109 @@
+option controls the distance measure used between genes. The possible values and
+their meanings are listed at the end of this document.
+
+```{r, message = FALSE}
+library("Biobase")
+```
+
+```{r}
+library("genefilter")
+data(sample.ExpressionSet)
+igenes <- c(300,333,355,419) ##the interesting genes
+closeg <- genefinder(sample.ExpressionSet, igenes, 10, method="euc", scale="none")
+names(closeg)
+```
````
Please combine into a single block as in the PDF.

devbird007 replied:
Done

jwokaty commented on vignettes/howtogenefinder.Rmd:
````diff
@@ -0,0 +1,109 @@
+genes (by index) for any of these by simply accessing the relevant component of
+`closeg`.
+
+```{r}
+closeg$"31539_r_at"
+```
+
+```{r}
+Nms1 <- featureNames(sample.ExpressionSet)[closeg$"31539_r_at"$indices]
+Nms1
+```
````
Combine into a single block


```{r}
closeg$"31539_r_at"
Nms1 <- featureNames(sample.ExpressionSet)[closeg$"31539_r_at"$indices]
Nms1

devbird007 replied:
Done

jwokaty commented:
@mcarlsn Thank you for reviewing. @devbird007 Please make these final changes. Provided that these changes are done and no further modifications are required, I will merge your PR. You can choose a new vignette to work on that is of medium or hard difficulty.

devbird007 committed:
Combined code chunks and labelled them

devbird007 committed:
named the code chunks

devbird007 commented:
@mcarlsn @jwokaty 
I have updated the name section to reflect only my name as you directed, I have replaced the HTML Zip file to reflect this change.

I also tried to name my code chunks, do let me know if the names are appropriate. I noticed that the previous Outreachy intern named her code chunks and I tried to imitate that.

jwokaty commented:
Just a little note. I noticed your indents and the old indents in the DESCRIPTION file were a little different, so I made this change afterward. Watch out for this in the future.

Your work has been merged into Bioconductor! Yay!

devbird007 commented:
> Just a little note. I noticed your indents and the old indents in the DESCRIPTION file were a little different, so I made this change afterward. Watch out for this in the future.
> 
> Your work has been merged into Bioconductor! Yay!

@jwokaty Alright I will, thank you so much haha😊
